### ISSUE TITLE
Crash in darkroom

### ISSUE THREAD

**Issue by kofa73** [darktable-org/darktable#16841]
### Describe the bug

I was working in the darkroom (either adjusting the perspective correction module or the contrast in filmic rgb), when darktable crashed. The backtrace file was empty; all I have is the console log from `darktable -d pipe` (that was the option I used the log a previous crash).
Today I edited many other files the same way, from the same camera, without a crash, so I think it won't be deterministic.

### Steps to reproduce

Get the raw and the sidecar from https://tech.kovacs-telekes.org/files/2024-05-20-darktable-crash/

### Where did you obtain darktable from?

downloaded from www.darktable.org

### darktable version

4.7.0+1257~g156a42d33a

### What OS are you using?

Linux

### What is the version of your OS?

Ubuntu 23.10

### Describe your system?

Ryzen 5 5600X

### Are you using OpenCL GPU in darktable?

Yes

### If yes, what is the GPU card and driver?

NVIDIA GeForce GTX 1060 6GB; 535.171.04

**Comment by jenshannoschwalm** [comment]
In the provided log we miss the "got details mask" ... could be the buffer data is wrong in this case or scaling/distorting of the mask failed. BTW no problem here so far.

**Comment by kofa73** [comment]
Thanks for looking into this. 

- This is the 2nd time my backtrace is empty. What could I do to prevent that? 
- What `-d` flags would provide the most useful output?

**Comment by jenshannoschwalm** [comment]
Likely `-d pipe -d verbose -d opencl`

**Comment by ralfbrown** [comment]
Given the choice between perspective and filmic, the fact that the pipe was running in fast mode indicates you were in perspective correction.

**Comment by jenshannoschwalm** [comment]
Yes, the perspective correction module was open/active. Still the processing was in diffuse and as some lines above would require the detail mask. This point was not reached so one idea would be the distorting of the detail mask somehow failed, the single modules distorting it including their roi would be shown via the -d verbose option

**Comment by jenshannoschwalm** [comment]
Sorry, don't understand Your question

**Comment by kofa73** [comment]
Since it seems the _filmic_ auto-picker area is off (it does not seem to handle the distortion introduced by _rotate and perspective_), maybe there's some other code that also does not handle it, leading to the crash.

**Comment by kofa73** [comment]
I'll share the logs when I encounter the next crash.

**Comment by jenshannoschwalm** [comment]
@kofa73 are you using keyboard shortcuts for rotation by any chance?

**Comment by kofa73** [comment]
> @kofa73 are you using keyboard shortcuts for rotation by any chance?

No.
![image](https://github.com/darktable-org/darktable/assets/3177739/5ca1b8fa-488d-4a58-b0c1-83d49f7e5960)
